cli/tests/pcluster/validators/test_all_validators.py

```diff
@@ -295,3 +309,296 @@ def test_validators_are_called_with_correct_argument(test_datadir, mocker):
     ebs_volume_size_snapshot_validator.assert_called()
     shared_ebs_volume_id_validator.assert_called()
     fsx_persistent_options_validator.assert_called()
+
+
+def test_scheduler_plugin_all_validators_are_called(test_datadir, mocker):
+    """Verify that all validators are called during validation."""
+    mockers = []
+    _mock_all_validators(mocker, mockers, additional_modules=[scheduler_plugin_validators])
+
+    # mock properties that use boto3 calls
+    mocker.patch(
+        "pcluster.config.cluster_config.HeadNode.architecture", new_callable=PropertyMock(return_value="x86_64")
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.SchedulerPluginComputeResource.architecture",
+        new_callable=PropertyMock(return_value="x86_64"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.HeadNodeNetworking.availability_zone",
+        new_callable=PropertyMock(return_value="us-east-1a"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.BaseClusterConfig.head_node_ami",
+        new_callable=PropertyMock(return_value="ami-12345678"),
+    )
+    mocker.patch(
+        "pcluster.config.cluster_config.SchedulerPluginClusterConfig.get_instance_types_data",
+    )
+    mocker.patch(
+        "pcluster.aws.ec2.Ec2Client.describe_image",
+        return_value=ImageInfo({"BlockDeviceMappings": [{"Ebs": {"VolumeSize": 35}}]}),
+    )
+
+    mock_aws_api(mocker)
+
+    # Need to load two configuration files to execute all validators because there are mutually exclusive parameters.
+    _load_and_validate(test_datadir / "scheduler_plugin_1.yaml")
+    _load_and_validate(test_datadir / "scheduler_plugin_2.yaml")
+
+    # Assert validators are called
+    for m in mockers:
+        if m["name"] in [
+            "TagKeyValidator",
+            "ClusterNameValidator",
+            "InstanceProfileValidator",
+            "RoleValidator",
+            "MixedSecurityGroupOverwriteValidator",
+            "HostedZoneValidator",
+            "InstanceTypeMemoryInfoValidator",
```

Comment:
Why are we skipping these 3 additional validators with respect to `test_all_validators_are_called()`?

Reply:
Because these validators are only call for scheduler plugin. For slurm schedulers, they are not called. `test_all_validators_are_called` are for slurm scheduler. Renamed the function